### PR TITLE
log_card_moved_to_complete

### PR DESCRIPTION
**Pull request by ThapeloMarakalla**
Related issues: [please specify]

## Description:

What are you up to? Fill us in :)

## I solemnly swear that:

- [x] My code follows the style guidelines of this project
- [x] I have merged the develop branch into my branch and fixed any merge conflicts
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Rokhuda — APPROVED** [review]

File: backend/curriculum_tracking/tests/test_activity_log_entry_creators.py [hunk 1]
```diff
@@ -155,8 +155,31 @@ def test_start_project(self):
 # class log_card_review_request_cancelled_Tests(TestCase):
 
 
-# class log_card_moved_to_complete_Tests(TestCase):
+class log_card_moved_to_complete_Tests(TestCase):
+    LIST_URL_NAME = "agilecard-list"
+    SUPPRESS_TEST_POST_TO_CREATE = True
```
Comment (sheenarbw) [inline backend/curriculum_tracking/tests/test_activity_log_entry_creators.py:161]: In you are4n't inheriting from the api test mixin then these lines aren't necessary

File: backend/curriculum_tracking/tests/test_activity_log_entry_creators.py [hunk 2]
```diff
@@ -155,8 +155,31 @@ def test_start_project(self):
...
+        actor_user = UserFactory(is_superuser=True)
+        card = AgileCardFactory(
+            status=AgileCard.IN_REVIEW,
+            content_item=ProjectContentItemFactory(
+                project_submission_type=ContentItem.LINK, template_repo=None
+            ),
+        )
+        self.login(actor_user)
 
+        add_review_url = f"{self.get_instance_url(card.id)}add_review/"
```
Comment (sheenarbw) [inline backend/curriculum_tracking/tests/test_activity_log_entry_creators.py:174]: Where is `self.get_instance_url` defined?  It's defined on a class that you didn't inherit from. So you will get an error here.

I don't think this test will pass. Are you sure that this test runs and that it passes?

File: backend/curriculum_tracking/tests/test_activity_log_entry_creators.py [hunk 3]
```diff
@@ -155,8 +155,31 @@ def test_start_project(self):
...
+        response = self.client.post(
+            add_review_url, data={"status": COMPETENT, "comments": "woohoo"}
+        )
+        self.assertEqual(response.status_code, 200)
+
+        card.refresh_from_db()
+        self.assertEqual(card.status, AgileCard.COMPLETE)
+        log_card_moved_to_complete.assert_called_with(card, actor_user)
```
Comment (sheenarbw) [inline backend/curriculum_tracking/tests/test_activity_log_entry_creators.py]: It's also important to make sure that log_card_moved_to_complete does the right thing. What is it meant to do and where are those tests?

**Review by sheenarbw — CHANGES_REQUESTED** [review]
These tests look incomplete. Also, wont they fail with a NameError? Did you run this?

File: backend/curriculum_tracking/tests/test_activity_log_entry_creators.py [hunk 4]
```diff
@@ -156,7 +156,31 @@ def test_start_project(self):
...
+        add_review_url = f"{self.get_instance_url(card.id)}add_review/"
+        response = self.client.post(
+            add_review_url, data={"status": COMPETENT, "comments": "woohoo"}
+        )
+        self.assertEqual(response.status_code, 200)
+
+        card.refresh_from_db()
+        self.assertEqual(card.status, AgileCard.COMPLETE)
+        self.assertEqual(LogEntry.objects.last().event_type.name, creators.CARD_MOVED_TO_COMPLETE)
+        self.assertEqual(LogEntry.objects.count(), 2)
```
Comment (Samantha-Hampton) [inline backend/curriculum_tracking/tests/test_activity_log_entry_creators.py:189]: I might be wrong but maybe to make this test complete and check that all the right things are happening with logging a card moving to complete be good to add assertions that the following are what they should be:

LogEntry.objects.last().object_1 == card.recruit_project
LogEntry.objects.last().object_2 == None
LogEntry.objects.last().actor_user == actor_user
LogEntry.objects.last().effected_user == card.assignee.first()
LogEntry.objects.last().timestamp == card.complete_time

(just double check that I've actually got those right, lol)

**Review by Samantha-Hampton — CHANGES_REQUESTED** [review]

File: backend/curriculum_tracking/tests/test_activity_log_entry_creators.py [hunk 5]
```diff
@@ -156,7 +156,36 @@ def test_start_project(self):
...
+
+        entry = LogEntry.objects.last()
+
+        card.refresh_from_db()
+        self.assertEqual(card.status, AgileCard.COMPLETE)
+        self.assertEqual(entry.event_type.name, creators.CARD_MOVED_TO_COMPLETE)
+        self.assertEqual(LogEntry.objects.count(), 2)
+        self.assertEqual(entry.effected_user, card.assignees.first())
+        self.assertEqual(entry.actor_user, actor_user)
+        self.assertEqual(entry.object_1, card.recruit_project)
```
Comment (Samantha-Hampton) [inline backend/curriculum_tracking/tests/test_activity_log_entry_creators.py:188]: The only things missing as per below are:

```
LogEntry.objects.last().object_2 == None
LogEntry.objects.last().timestamp == card.complete_time
```

Perhaps it's overkill? If you think so I'll re-review as approved and then we can see what Sheena thinks 🤷🏼‍♀️ Let me know :)

**Review by Samantha-Hampton — CHANGES_REQUESTED** [review]

File: backend/curriculum_tracking/tests/test_activity_log_entry_creators.py (hunk 5 above)
Comment (Andy-Nkumane) [reply backend/curriculum_tracking/tests/test_activity_log_entry_creators.py:188]: there is a small issue with the timestamp and the complete time - they differ with the split seconds

Comment (Samantha-Hampton) [reply backend/curriculum_tracking/tests/test_activity_log_entry_creators.py:188]: ah ok, fair enough :)

**Review by Samantha-Hampton — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**